src/blocks/map/edit-gmaps-with-api-key.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * External dependencies
+ */
+import { GoogleMap, Marker, useLoadScript } from '@react-google-maps/api';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import { Spinner } from '@wordpress/components';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import GMapStyles from './map-styles';
+
+const GoogleMapWithApiKey = ( { apiKey, props } ) => {
+	const { isLoaded } = useLoadScript( {
+		googleMapsApiKey: apiKey,
+	} );
+
+	const {
+		attributes,
+	} = props;
+
+	const {
+		address,
+		fullscreenControl,
+		iconSize,
+		mapTypeControl,
+		skin,
+		streetViewControl,
+		zoom,
+		zoomControl,
+	} = attributes;
+
+	const [ coords, setCoords ] = useState( null );
+
+	useEffect( () => {
+		if ( isLoaded ) {
+			const geocoder = new window.google.maps.Geocoder();
+			geocoder.geocode(
+				{ address },
+				function( results, status ) {
```

Comment:
I think you can use an arrow function here instead. Would remove the `.bind(this)`

Reply:
Oh yeah, would be a lot cleaner, agreed ;)